### PR TITLE
[settings] fix seq fault if addon use same setting in different groups

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
Related to https://pastebin.com/nEa0TQBc and https://forum.kodi.tv/showthread.php?tid=298462&pid=2610026#pid2610026

Crash was from settings.xml here: https://github.com/Zomboided/service.vpn.manager/blob/master/resources/settings.xml

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/settings/lib/SettingsManager.cpp [hunk 1]
```diff
@@ -1248,7 +1248,9 @@ void CSettingsManager::ResolveReferenceSettings(std::shared_ptr<CSettingSection>
         else
         {
           referencedSetting = itReferencedSetting->second.setting;
-          m_settings.erase(FindSetting(referenceSetting->GetId()));
+          auto itReferencedSetting2 = FindSetting(referenceSetting->GetId());
```
Comment (Montellese) [inline xbmc/settings/lib/SettingsManager.cpp]: You could reuse the existing `itReferencedSetting` variable.

Comment (AlwinEsch) [inline xbmc/settings/lib/SettingsManager.cpp]: Oh yes, you are right. Becomes changed.

Is this a fix enough, or maybe another part who need change?

Comment (Montellese) [inline xbmc/settings/lib/SettingsManager.cpp]: This is ok for solving the crash if the same setting is used more than twice (in which case it has already been removed from `m_settings`. There are still issues however with reference settings and enable/visibility conditions which I'm trying to solve.

**Review by Montellese — APPROVED** [review]
Change the local variable being used and then this is good to go in. Thanks for looking into it.

**Pull request merged**